code/src/layouts/header/components/header-toolbar/HeaderToolbar.tsx

```diff
@@ -56,19 +64,50 @@ const HeaderToolbar = () => {
     <AppTypography variant="caption-small">{itemsInCartCount}</AppTypography>
   );
 
-  const AuthButton = isAuthenticated ? (
-    <AppButton onClick={handleLogout} variant="danger" size="small">
+  const loadingButton = isLoadingAuth ? <AppLoader /> : null;
+
+  const logoutButton = isAuthenticated ? (
+    <AppButton
+      onClick={handleLogout}
+      variant="danger"
+      size="small"
+      data-testid="LogoutButton"
+    >
       <LogoutIcon />
     </AppButton>
-  ) : (
-    <AppButton onClick={handleOpenAuthModal}>
-      <AppTypography translationKey="signIn.label" />
-    </AppButton>
-  );
+  ) : null;
```

Comment:
Why did you choose to use ternaries and not `&&`?

Reply:
> Why did you choose to use ternaries and not `&&`?

Fixed